File: file.py
# Function to append text to an existing file entry
def append_entry():
    name = get_file_name()
    filename = name + ".txt"

    with open(filename, "a") as file_management_system:
        print(f"Type your file content for {name} (Enter END on a new line when done.):")
        while True:
            line = input()
            if line == "END":
                break
            file_management_system.write(line + "\n")


# Function to create a new file entry
def create_entry():
    name = get_file_name()
    filename = name + ".txt"

    print(f"Type your file content for {name} (Enter END on a new line when done.):")
    with open(filename, "w") as file_management_system:
        while True:
            line = input()
            if line == "END":
                break
            file_management_system.write(line + "\n")


# Function to read an entry file and print that file to the terminal
def read_entry():
    name = get_file_name()
    filename = name + ".txt"

    try:
        with open(filename, "r") as file_management_system:
            for line in file_management_system:
                print(line.rstrip())
    except FileNotFoundError:
        print(f"Could not find file entry for {name}")


def get_file_name():
    name = input("Please enter the file name without white spaces: ")
    return name


if __name__ == "__main__":
    print("Welcome to the File Management System!")

    # While loop for operations, run until exit or End-Of-File (EOF)
    while True:
        print("\nChoose an operation:")
        print("(x) : Exit")
        print("(n) : Create a new file")
        print("(r) : Read a file")
        print("(a) : Append to an existing file")

        choice = input()

        # Check if the user wants to exit
        if choice == 'x':
            # Return from program
            break

        switcher = {
            'n': create_entry,
            'r': read_entry,
            'a': append_entry
        }

        # Get the function from switcher dictionary
        func = switcher.get(choice, lambda: print("ERROR: Invalid input. Choose again."))

        # Execute the function
        func()

File: file.cpp
#include <fstream>
#include <iostream>
#include <limits>
#include <string>
using namespace std;


// Function declarations
void appendEntry();
void createEntry();
void readEntry();
string getFileName();


/**
 * The main function will loop through the file management system
*/
int main()
{
    cout << "Welcome to the File Management System!" << endl;

    // While loop for operations, run until exit or End-Of-File (EOF)
    while (true)
    {
        cout << endl;
        cout << "Choose an operation:" << endl;
        cout << "(x) : Exit" << endl;
        cout << "(n) : Creat a New File" << endl;
        cout << "(r) : Read a File" << endl;
        cout << "(a) : Append to an existing file" << endl;

        char choice;
        cin >> choice;

        // Check if the user wants to exit
        if (!cin || choice == 'x')
        {
            // Return from program
            break;
        }

        // Reset the state of the cin stream
        cin.clear();
        cin.ignore(numeric_limits<streamsize>::max(), '\n');

        switch (choice)
        {
        case 'n':
            createEntry();
            break;
        case 'r':
            readEntry();
            break;
        case 'a':
            appendEntry();
            break;
        default:
            cout << "ERROR: Invalid input. Choose again." << endl;
            break;
        }
    }

    return 0;   // Return 0 on success
}


/**
 * This function will append text to an existing file entry
*/
void appendEntry()
{
    string name = getFileName();
    string filename = name + ".txt";

    ofstream fileManagementSystem;
    fileManagementSystem.open(filename, ofstream::app); // Open the file for appending new texts

    cout << "Type your file name for " << name << " (Enter END on a new line when done.):" << endl;

    string line;
    while (getline(cin, line))
    {
        if (line == "END")
        {
            break;
        }
        fileManagementSystem << line << endl;
    }

    fileManagementSystem.close();
}


/**
 * This function will create a new file entry
*/
void createEntry()
{
    string name = getFileName();
    string filename = name + ".txt";

    cout << "Type your file name for " << name << " (Enter END on a new line when done.):" << endl;

    ofstream fileManagementSystem(filename);

    string line;
    while (getline(cin, line))
    {
        if (line == "END")
        {
            break;
        }
        fileManagementSystem << line << endl;
    }

    fileManagementSystem.close();
}


/**
 * This function will read an entry file and print that file to the terminal
*/
void readEntry()
{
    string name = getFileName();
    string filename = name + ".txt";

    // TODO: Write your code here.
    ifstream fileManagementSystem;
    fileManagementSystem.open(filename); // open the file

    if (!fileManagementSystem)
    {
        cout << "Could not find file entry for " << name << endl;
        return;
    }

    string line;
    while (getline(fileManagementSystem, line))
    {
        cout << line << endl;
    }

    fileManagementSystem.close();
}


string getFileName()
{
    string name;
    cout << "Please enter the file name without white spaces: ";
    cin >> name;

    // Return the date
    return name;
}
